### ISSUE TITLE
Polls enablement/backport?

### ISSUE THREAD

**Issue by krc** [johnHackworth/goblin#27]
Looks like Firefish has polls, did they implement them after Goblin was forked, or is it like an option to turn on in server config?

Looks like they allow more than the 4 options that Mastodon does, and allow voting for multiple option (if you enable it at poll creation time).

**Comment by sirilyan** [comment]
Note: Tumblr cross-combat isn't possible - the public API doesn't permit voting in polls or getting results.

**Comment by johnHackworth** [comment]
yup, this is on my radar since time ago, but they would be 100% independent polls than the ones from tumblr I'm afraid

**Comment by sirilyan** [comment]
So the RSS for a Tumblr post that contains a poll _does_ have a div with a `data-npf` attribute, and that attribute contains the poll object:

```json
{
  "type": "poll",
  "client_id": "3f149487-7424-42eb-81c8-c2ec352704ee",
  "question": "no nuance. what do you have",
  "answers": [
    {
      "client_id": "9980ddd7-7a50-49fb-91e0-ed229a0e4c22",
      "answer_text": "small boobs"
    },
    {
      "client_id": "61ec4418-96ed-4ddf-afb2-cef46302e1fc",
      "answer_text": "big boobs"
    }
  ],
  "settings": {
    "multiple_choice": false,
    "close_status": "closed-after",
    "expire_after": 604800,
    "source": "tumblr"
  }
}
```

(I love how every single time I find an example of anything it's something very, very Tumblr.)

So although it's impossible to use the API, it might be possible to make Tumblr poll posts look nicer and offer a link to Tumblr if you want to vote. The markup could look something like this:

```html
<p>Blah blah poll title</p>
<ul>
  <li>Option 1</li>
  <li>Option 2</li>
</ul>
<p>
  <i>Note: You cannot vote on Tumblr polls from goblin.</i>
  <a href="#">Go to the original post.</a>
</p>
```
